enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,668 @@
+                "openshift/conformance"
+            ],
+
+            # Test cases can be advertised from across a number
+            # of extension binaries. The following cel expressions
+            # are OR'd together. If they select a test advertised
+            # from another binary, they will be considered part of
+            # this advertised suite.
+            # Test tags, names, and other attributes will be
+            # described later.
+            "qualifiers": [
+                "(test.tags.suite==\"fips\" || test.name.contains(\"fips\")) && !test.labels.has(\"Disruptive\")"
+            ]
+        }
+    ]
+    
+}
+```
+
+##### List - Extension Test Listing
+The "list" verb will output information about tests exposed by the extension
+for different environments. A test "environment" describes the attributes of 
+cluster under test (e.g. the cloud provider, the topology, etc).
+
+The "list" verb should eliminate tests that are not appropriate for the
+specified environment. 
+
+The arguments accepted by the verb may vary over time and will be
+defined by the extension interface version implemented by the binary.
+
+**Version 1**
+```
+$ extension-binary list 
+  --platform      The hardware or cloud platform ("aws", "gcp", "metal", ...).
+  --network       The network of the target cluster ("ovn", "sdn"). 
+  --upgrade       The upgrade that was performed prior to the test run ("micro", "minor").
+  --topology      The target cluster topology ("ha", "microshift", ...).
+  --architecture  The CPU architecture of the target cluster ("amd64", "arm64").
+  --installer     The installer used to create the cluster ("ipi", "upi", "assisted", ...).
+  --config        The component configuration to assume is active on the cluster.
+  --version       "major.minor" version of target cluster. 
+```
+
+The "list" verb will not be provided a kubeconfig and must output tests based solely
+on environment information. If no environment arguments are provided, all tests must be listed.
+
+Listings are formatted as JSONL with one test description per line. A full listing contains
+zero or more test description objects. The following example shows an abbreviated JSONL listing 
+for three tests.
+```python
+{ "name": "T1", }
+{ "name": "T2", }
+{ "name": "T3", }
+```
+
+A test description object contains a number of attributes to inform `openshift-tests`
+discovery and planning. Note that the following annotated example would be encoded into a single 
+line of output in actual listing output.
+```python
+{
+  
+  # Base, human-readable test name. 
+  "name": "openssl version compliance",
+
+  # If a test name is updated at any time in the future,
+  # originalName must report the original name of the 
+  # testing logic. This allows component readiness
+  # to display the human-readable version of the test
+  # name while considering test runs across name changes.
+  "originalName": "security version compliance",

```

Comment:
I really like baking this into the component repos, might get us more renames as very very few people go to the mapping repo.

It would be neat if something could comment on PRs where we see added+removed tests that "if this was a rename, please do ...". That would be relatively often a rename. Problem for another day though.